ab2d-bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDSearchImplTest.java

```diff
@@ -0,0 +1,137 @@
+
+import ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException;
+import gov.cms.ab2d.fhir.FhirVersion;
+
+class BFDSearchImplTest {
+
+  private static int patientId = 1;
+  private static OffsetDateTime since = OffsetDateTime.now();
+  private static OffsetDateTime until = OffsetDateTime.now();
+  private static int pageSize = 10;
+  private static String bulkJobId = "bulkJobId";
+  private static FhirVersion version = FhirVersion.R4;
+  private static String contractNum = "contractNum";
+  private static String filePath = "src/test/resources/bb-test-data/eob/20010000001115.json";
+  private static String activeProfile = "test";
+  private static String bfdUrl = "http://localhost:8080";
+
+  HttpClient httpClient200() throws IOException {
+    HttpClient httpClient = mock(HttpClient.class);
+    CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+    when(httpClient.execute(any())).thenReturn(response);
+    when(response.getStatusLine()).thenReturn(mock(org.apache.http.StatusLine.class));
+    when(response.getStatusLine().getStatusCode()).thenReturn(HttpStatus.SC_OK);
+    when(response.getEntity()).thenReturn(mock(org.apache.http.HttpEntity.class));
+    String fileContent = Files.readString(Paths.get(filePath));
+    when(response.getEntity().getContent()).thenReturn(new ByteArrayInputStream(fileContent.getBytes()));
+    return httpClient;
+  }
+
+  HttpClient httpClient404() throws IOException {
+    HttpClient httpClient = mock(HttpClient.class);
+    CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+    when(httpClient.execute(any())).thenReturn(response);
+    when(response.getStatusLine()).thenReturn(mock(org.apache.http.StatusLine.class));
+    when(response.getStatusLine().getStatusCode()).thenReturn(HttpStatus.SC_NOT_FOUND);
+    return httpClient;
+  }
+
+  HttpClient httpClient500() throws IOException {
+    HttpClient httpClient = mock(HttpClient.class);
+    CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+    when(httpClient.execute(any())).thenReturn(response);
+    when(response.getStatusLine()).thenReturn(mock(org.apache.http.StatusLine.class));
+    when(response.getStatusLine().getStatusCode()).thenReturn(HttpStatus.SC_INTERNAL_SERVER_ERROR);
+    return httpClient;
+  }
+
+  @Test
+  void testSearchEOB200() throws IOException{
+    // Setup mocks
+    HttpClient httpClient = httpClient200();
+    Environment environment = mock(Environment.class);
+    when(environment.getActiveProfiles()).thenReturn(new String[] { activeProfile });
+
+    // Setup classes
+    BfdClientVersions bfdClientVersions = new BfdClientVersions(bfdUrl, httpClient);
+    BFDSearchImpl bfdSearchImpl = new BFDSearchImpl(httpClient, environment, bfdClientVersions);
+
+    // Business logic & assertion(s)
+    IBaseBundle result = bfdSearchImpl.searchEOB(patientId, since, until, pageSize, bulkJobId, version, contractNum);
+    assertNotNull(result);
```

Comment:
I tried to add another assertion here where I assert on the request URL for the HTTP client, but I couldn't get that to work.

This code:

```
    verify(httpClient).execute(new HttpGet("http://localhost:8080/v2/fhir/ExplanationOfBenefit?patient=1&excludeSAMHSA=true&_count=10"));
```

Gave me this error:

```
BFDSearchImplTest > testSearchEOB200() FAILED
    Argument(s) are different! Wanted:
    httpClient.execute(
        GET http://localhost:8080/v2/fhir/ExplanationOfBenefit?patient=1&excludeSAMHSA=true&_count=10 HTTP/1.1
    );
    -> at gov.cms.ab2d.bfd.client.BFDSearchImplTest.testSearchEOB200(BFDSearchImplTest.java:83)
    Actual invocations have different arguments:
    httpClient.execute(
        GET http://localhost:8080/v2/fhir/ExplanationOfBenefit?patient=1&excludeSAMHSA=true&_count=10 HTTP/1.1
    );
    -> at gov.cms.ab2d.bfd.client.BFDSearchImpl.getEOBSFromBFD(BFDSearchImpl.java:90)
        at app//gov.cms.ab2d.bfd.client.BFDSearchImplTest.testSearchEOB200(BFDSearchImplTest.java:83)
```

It looks the same, but says they are different :( probably something to do with comparing two instances of a class with each other.

Reply:
its interesting why V1 works but V2 fails. Have this code in place and comment it  with notes that couldnt get it to work for V2 as TODO?